src/Decapodes2/simulation.jl

```diff
@@ -141,6 +141,7 @@ gensim(d::AbstractNamedDecapode) = gensim(d,
     vcat(collect(infer_state_names(d)), d[incident(d, :Literal, :type), :name]))
 
 function gensim(d::AbstractNamedDecapode, input_vars)
+  recognize_types(d)
```

Comment:
This looks mutating, should have ! In the name

Reply:
If a function calls a mutating function on its argument, then it is mutating too. What is `recognize_types` adding over `infer_types!`